server/src/com/thoughtworks/go/server/service/PipelineConfigService.java

```diff
@@ -184,7 +184,7 @@ public void onPipelineConfigChange(PipelineConfig pipelineConfig, String group)
 
     @Override
     public void onConfigChange(CruiseConfig newCruiseConfig) {
-        PipelineConfigurationCache.getInstance().onConfigChange(goConfigService.getConfigForEditing());
+        PipelineConfigurationCache.getInstance().onConfigChange(goConfigService.cruiseConfig());
```

Comment:
Removing usages of `getConfigForEditing` might not have helped avoid this bug. `getConfigForEditing` was deprecated as we did not want to expose the the full cruise-config object. We would have gone with pulling out the required object from this and pass it along. However, in this case, the object to be passed along was the preprocessed version of the config, ie. with template & params inlined and we were returning the other one(without inlining).
However, I agree that we should remove usages of deprecated methods in general and delete them altogether.